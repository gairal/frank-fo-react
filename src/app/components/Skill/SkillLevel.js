import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { CircularProgress } from '@material-ui/core';

export default class SkillLevel extends Component {
  static propTypes = {
    level: PropTypes.number.isRequired,
  };

  constructor(props) {
    super(props);

    this.state = {
      progress: 0,
    };

    this.interval = null;
  }

  componentDidMount() {
    this.animateLevel();
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  updateLevel(resolve) {
    const { level } = this.props;
    const { progress } = this.state;
    if (progress >= level) {
      clearInterval(this.interval);
      resolve(progress);
    } else {
      this.setState(prevState => ({
        progress: prevState.progress + 5,
      }));
    }
  }

  animateLevel() {
    return new Promise(resolve => {
      this.interval = setInterval(() => this.updateLevel(resolve), 5);
    });
  }

  render() {
    const { progress } = this.state;
    return <CircularProgress variant="static" value={progress} thickness={5} />;
  }
}
