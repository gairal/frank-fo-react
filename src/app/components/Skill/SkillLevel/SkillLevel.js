import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { CircularProgress } from 'material-ui/Progress';

export default class SkillLevel extends Component {
  static propTypes = {
    level: PropTypes.number.isRequired,
  }

  constructor(props) {
    super(props);

    this.state = {
      progress: 0,
    };
  }

  componentDidMount() {
    this.interval = setInterval(() => {
      if (this.state.progress >= this.props.level) {
        clearInterval(this.interval);
      } else {
        this.setState(prevState => ({
          progress: prevState.progress + 5,
        }));
      }
    }, 5);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    return (
      <CircularProgress mode="determinate" value={this.state.progress} thickness={5} />
    );
  }
}
