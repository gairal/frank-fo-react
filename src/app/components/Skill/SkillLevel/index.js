import skillLevel from './SkillLevel';

export default skillLevel;
