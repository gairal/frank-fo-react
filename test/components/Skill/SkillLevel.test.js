import React from 'react';
import renderer from 'react-test-renderer';
import SkillLevel from '@/components/Skill/SkillLevel';

describe('components > Skill > SkillLevel', () => {
  it('Renders correctly default', () => {
    const component = renderer.create(<SkillLevel level={20} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
