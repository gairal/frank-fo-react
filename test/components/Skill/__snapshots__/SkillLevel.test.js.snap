// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Skill > SkillLevel Renders correctly default 1`] = `
<div
  aria-valuenow={0}
  className="MuiCircularProgress-root-1 MuiCircularProgress-colorPrimary-4 MuiCircularProgress-static-2"
  role="progressbar"
  style={
    Object {
      "height": 40,
      "transform": "rotate(-90deg)",
      "width": 40,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg-6"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle-7 MuiCircularProgress-circleStatic-8"
      cx={44}
      cy={44}
      fill="none"
      r={19.5}
      strokeWidth={5}
      style={
        Object {
          "strokeDasharray": "122.522",
          "strokeDashoffset": "122.522px",
        }
      }
    />
  </svg>
</div>
`;
